### ISSUE TITLE
Hetzner Failed to get node infos for groups

### ISSUE THREAD

**Issue by timowevel1** [kubernetes/autoscaler#5185]
**Which component are you using?**:
Rancher with K3S 1.24
Autoscaler v1.25.0
Hetzner Cloud

cluster-autoscaler

**What version of the component are you using?**:
1.25.0

**What did you expect to happen?**:
Not sure, it doesnt throw errors on initial bootup? Not sure whats the issue there

`root@Rancher1:~# kubectl logs cluster-autoscaler-6bbc7d777-65pmm --namespace=kube-system
I0911 19:39:20.180846       1 leaderelection.go:248] attempting to acquire leader lease kube-system/cluster-autoscaler...
I0911 19:39:37.741205       1 leaderelection.go:258] successfully acquired lease kube-system/cluster-autoscaler
W0911 19:39:39.720046       1 hetzner_servers_cache.go:94] Fetching servers from Hetzner API
I0911 19:39:39.912952       1 node_instances_cache.go:156] Start refreshing cloud provider node instances cache
I0911 19:39:39.913027       1 node_instances_cache.go:168] Refresh cloud provider node instances cache finished, refresh took 28.352µs
I0911 19:39:49.914012       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:39:49.914795       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:39:49.916052       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
I0911 19:39:59.917076       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:39:59.917134       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:39:59.917482       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
I0911 19:40:09.918765       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:40:09.918817       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:40:09.919155       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://rancher2 exists error: failed to get servers for node rancher2 error: server not found
I0911 19:40:19.919815       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:40:19.919862       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:40:19.920137       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://rancher3 exists error: failed to get servers for node rancher3 error: server not found
I0911 19:40:29.921380       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:40:29.921413       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:40:29.921692       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
W0911 19:40:39.922774       1 hetzner_servers_cache.go:94] Fetching servers from Hetzner API
I0911 19:40:40.191242       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:40:40.191356       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:40:40.191946       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://rancher3 exists error: failed to get servers for node rancher3 error: server not found
I0911 19:40:50.192858       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:40:50.193165       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:40:50.193636       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
I0911 19:41:00.194803       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:41:00.194886       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:41:00.195563       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
I0911 19:41:10.196672       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:41:10.196707       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:41:10.197092       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://rancher3 exists error: failed to get servers for node rancher3 error: server not found
I0911 19:41:20.197971       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
I0911 19:41:20.198032       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
E0911 19:41:20.198962       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
I0911 19:41:30.199314       1 hetzner_node_group.go:437] Set node group pool1 size from 0 to 0, expected delta 0
I0911 19:41:30.199365       1 hetzner_node_group.go:437] Set node group draining-node-pool size from 0 to 0, expected delta 0
E0911 19:41:30.199588       1 static_autoscaler.go:298] Failed to get node infos for groups: failed to check if server k3s://worker-1 exists error: failed to get servers for node worker-1 error: server not found
`

**How to reproduce it (as minimally and precisely as possible)**:
Using 1.25.0

When I use 1.23.1 I dont see the errors, but also it doesnt scale

**Anything else we need to know?**:

I got  following servers running:
Rancher1, Rancher2, Rancher3 as Control Planes
worker1 as worker  node, all inside the cluster, I tried to deploy the autoscaler with the example yaml in the Hetzner folder.

My nodes in the cluster:
![image](https://user-images.githubusercontent.com/25285295/189546240-bb4de452-83b6-4df3-8602-21990faedbcc.png)

In Hetzner:
![image](https://user-images.githubusercontent.com/25285295/189546247-a1ca79ed-889e-4032-88ee-dbf1d5d29a11.png)

**Comment by gigipl** [comment]
Same here.

**Comment by tomasnorre** [comment]
I'm facing the same problem.

**Comment by jensjeflensje** [comment]
Hi, I ran into the same issue and fixed it by using the 1.24 autoscaler. I'm also running k3s 1.24 and matching that version with the autoscaler version, it works. You just have to check out the `cluster-autoscaler-release-1.24` version of autoscaler, and build and use that image.

**Comment by MarshmallowSoup** [comment]
Hello, I have a similar issue. My workers are creating from the ready snapshot and it takes approximately 15 minutes to create it and autoscaler gives an error `failed to start server win-worker-7520785252293546637 error: timeout waiting for server win-worker-7520785252293546637 `. This occurs before the server is even created, which causes the autoscaler to remove the server and create a new one, resulting in the same timeout problem.

**Comment by MarshmallowSoup** [comment]
Same here

**Comment by apricote** [comment]
/area provider/hetzner

It looks like you are using a different cloud-controller-manager than `hcloud-cloud-controller-manager` (HCCM). The Hetzner Cluster Autoscaler Provider expects that the Node.ProviderID matches the format HCCM (`hcloud://$SERVER_ID`), and if it does not then it can not properly resolve the Nodes.

**Comment by Mrono** [comment]
Running into this with 1.28 k3s and the chart for cluster-autoscaler-9.36.0

**Comment by apricote** [comment]
Hey @Mrono,

have you checked my comment above and compared against your nodes?

> It looks like you are using a different cloud-controller-manager than hcloud-cloud-controller-manager (HCCM). The Hetzner Cluster Autoscaler Provider expects that the `Node.ProviderID` matches the format HCCM (`hcloud://$SERVER_ID`), and if it does not then it can not properly resolve the Nodes.

**Comment by apricote** [comment]
@Bryce-Soghigian Not sure if you reopened because you experience the issue or for issue maintenance reasons. 

I do believe that this can be closed. I have provided an explanation for the issue the user was facing [5 months ago](https://github.com/kubernetes/autoscaler/issues/5185#issuecomment-1772133175). I have kept it open to allow the affected users to respond whether it fixed their problem, but so far no one has confirmed or denied that the suggested fix works.

Not sure what to do besides letting it be closed through the lifecycle or closing it myself.
